easybuild/easyblocks/c/cp2k.py

```diff
@@ -797,11 +797,21 @@ def install_step(self):
         targetdir = os.path.join(self.installdir, 'bin')
         copy_dir(exedir, targetdir)
 
-        # copy libraries
+        # copy libraries and include files, not sure what is strictly required so we take everything
         if self.cfg['library']:
             libdir = os.path.join(self.cfg['start_dir'], 'lib', self.typearch, self.cfg['type'])
             targetdir = os.path.join(self.installdir, 'lib')
             copy_dir(libdir, targetdir)
+            # Also need to populate the include directory
+            targetdir = os.path.join(self.installdir, 'include')
+            mkdir(targetdir)  # Needs to exist for copy below to work correctly
+            include_file = os.path.join(self.cfg['start_dir'], 'src', 'start', 'libcp2k.h')
+            copy_file(include_file, targetdir)
```

Comment:
@ocaisa `copy_file` will make sure the target dir exists, so you can shorten this to:

```python
target_inc_dir = os.path.join(self.installdir, 'include')
libcp2k_header = os.path.join(self.cfg['start_dir'], 'src', 'start', 'libcp2k.h')
copy_file(libcp2k_header, target_inc_dir)

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
Yeah, pretty sure I am right here. `copy_file` creates the parent directory of the target but it is working under the assumption that the target is a file. I'm abusing it here since I know it uses `copy2` under the hood and `copy2` allows me to use a directory as a target but I need to make sure that directory exists first (otherwise it copies it as a file with the directory name).

Reply:
@ocaisa Then just copy to `os.path.join(targetdir, os.path.basename(include_file))` instead?